scripts/core/itemList/items-list-limited.tsx

```diff
@@ -14,31 +15,47 @@ interface IProps {
 
 interface IState {
     items: Array<IArticle>;
+    relatedEntities: IRelatedEntities;
 }
 
 // DOES NOT SUPPORT item actions, pagination, multiselect
 class ItemsListLimitedComponent extends React.Component<IProps, IState> {
     monitoringState: any;
+    private abortController: AbortController;
 
     constructor(props: any) {
         super(props);
 
         this.state = {
             items: null,
+            relatedEntities: {},
         };
 
         this.monitoringState = ng.get('monitoringState');
+        this.abortController = new AbortController();
     }
     componentDidMount() {
         const {ids} = this.props;
 
         this.monitoringState.init().then(() => {
             Promise.all(ids.map((id) => dataApi.findOne<IArticle>('search', id)))
                 .then((items) => {
-                    this.setState({items});
+                    getRelatedEntities(
+                        items,
+                        this.state.relatedEntities,
+                        this.abortController.signal,
+                    ).then((relatedEntities) => {
+                        this.setState({
+                            items,
+                            relatedEntities: mergeRelatedEntities(this.state.relatedEntities, relatedEntities),
```

Comment:
seems like there is always merge happening after return from `getRelatedEntities`, given that those are available in the call I would probably rather do the merge inside `getRelatedEntities` before returning